fvtest/compilerunittest/p/BinaryEncoder.cpp

```diff
@@ -0,0 +1,2856 @@
+
+        BinaryInstruction instr(_words);
+
+        for (size_t i = 0; i < _words.size(); i++)
+            instr._words[i] ^= other._words[i];
+
+        return instr;
+    }
+
+    bool operator==(const BinaryInstruction& other) const {
+        return other._words == _words;
+    }
+};
+
+std::ostream& operator<<(std::ostream& os, const BinaryInstruction& instr) {
+    os << "[";
+
+    for (size_t i = 0; i < instr.words().size(); i++) {
+        os << " 0x" << std::hex << std::setw(8) << std::setfill('0') << instr.words()[i];
+    }
+
+    os << " ]";
+
+    return os;
+}
+
+std::ostream& operator<<(std::ostream& os, const TR::InstOpCode::Mnemonic& opCode) {
+    os << TR::InstOpCode::metadata[opCode].name;
+    return os;
+}
+
+std::ostream& operator<<(std::ostream& os, const TR::RealRegister::RegNum& reg) {
+    if (reg >= TR::RealRegister::FirstGPR && reg <= TR::RealRegister::LastGPR)
+        os << "gr" << (static_cast<int>(reg) - static_cast<int>(TR::RealRegister::FirstGPR));
+    else if (reg >= TR::RealRegister::FirstFPR && reg <= TR::RealRegister::LastFPR)
+        os << "fp" << (static_cast<int>(reg) - static_cast<int>(TR::RealRegister::FirstFPR));
+    else if (reg >= TR::RealRegister::FirstVRF && reg <= TR::RealRegister::LastVRF)
+        os << "vr" << (static_cast<int>(reg) - static_cast<int>(TR::RealRegister::FirstVRF));
+    else if (reg >= TR::RealRegister::FirstCCR && reg <= TR::RealRegister::LastCCR)
+        os << "cr" << (static_cast<int>(reg) - static_cast<int>(TR::RealRegister::FirstCCR));
+    else
+        os << "???";
+
+    return os;
+}
+
+class PowerBinaryEncoderTest : public TRTest::CodeGenTest {
+public:
+    TR::Node* fakeNode;
+
+    uint32_t buf[64];
+
+    TR::RealRegister* createReg(TR_RegisterKinds kind, TR::RealRegister::RegNum reg, TR::RealRegister::RegMask mask) {
+        return new (cg()->trHeapMemory()) TR::RealRegister(kind, 0, TR::RealRegister::Free, reg, mask, cg());
+    }
+
+    PowerBinaryEncoderTest() {
+        fakeNode = TR::Node::create(TR::treetop);
+        cg()->setBinaryBufferStart(reinterpret_cast<uint8_t*>(&buf[0]));
+    }
+
+    BinaryInstruction getEncodedInstruction(size_t size, uint32_t off = 0) {
+        return std::vector<uint32_t>(&buf[off], &buf[off + (size / 4)]);
+    }
+
+    BinaryInstruction encodeInstruction(TR::Instruction *instr, uint32_t off = 0) {
+        instr->estimateBinaryLength(0);
+        cg()->setBinaryBufferCursor(reinterpret_cast<uint8_t*>(&buf[0]) + off);
+
+        instr->generateBinaryEncoding();
+        return getEncodedInstruction(instr->getBinaryLength(), off);
+    }
+
+    BinaryInstruction getBlankEncoding(TR::InstOpCode opCode) {
+        reinterpret_cast<uint32_t*>(opCode.copyBinaryToBuffer(reinterpret_cast<uint8_t*>(&buf[0])));
+
+        return getEncodedInstruction(4);
+    }
+
+    BinaryInstruction findFlippedBits(TR::InstOpCode op1, TR::InstOpCode op2) {
```

Comment:
Howabout `findDifferentBits`:
```suggestion
    BinaryInstruction findDifferentBits(TR::InstOpCode op1, TR::InstOpCode op2) {
```

Reply:
Hmm... I'm not sure that `findDifferentBits` really clarifies the meaning here... Maybe we could call this `findDifferingBits` instead?

Reply:
Yes. `findDifferingBits` is definitely clearer.

Reply:
Changed in 3c75b24c0c32f8aba54383c21b0b455a2f52ea7d